governance/engine.go

```diff
@@ -495,7 +498,7 @@ func (e *Engine) intoToSubmit(ctx context.Context, p *types.Proposal) (*ToSubmit
 		enactTime := time.Unix(p.Terms.EnactmentTimestamp, 0)
 		newMarket := p.Terms.GetNewMarket()
 
-		auctionDuration := enactTime.Sub(closeTime)
+		auctionDuration := time.Duration((enactTime.Unix() - closeTime.Unix()) * int64(time.Second))
```

Comment:
Minor: I'm sure theres a good reason you're stripping off all the precision but given that `enactTime` is already just in precision seconds, I think all you need to do is here is`closeTime := e.timeService.GetTimeNow().Truncate(time.Second)`

Also that `FIXME` comment about the auctionState, do we not also need to trunctate the time passed into `NewAuctionState`?

Reply:
addressed